### PR TITLE
sql: add smallint

### LINKED ISSUE #7416: sql: fully support int2
`int2`/`smallint` are aliases for `int4`, but should be their own type; this is a sub-issue of #5169

### PR DESCRIPTION
**Pull request by sploiselle**
Closes #7416

### CODE REVIEW

File: test/sqllogictest/chbench.slt [hunk 1]
```diff
@@ -300,23 +300,19 @@ ORDER BY n_name, su_name, i_id
...
-| ArrangeBy (#0) (#2)
+| ArrangeBy (#0)
 
 %8 =
 | Get materialize.public.region (u37)
 | ArrangeBy (#0)
 
 %9 =
-| Join %5 %6 %7 %8 (= #17 #18) (= #21 #25) (= #27 #29)
-| | implementation = DeltaQuery
```
Comment (mjibson) [inline test/sqllogictest/chbench.slt:315]: Is this ok? I don't know anything besides I think DeltaQuery is good?

**Review by mjibson — COMMENTED** [review]
Can you run `cargo test -p coord verify_function_sanity -- --ignored` to ensure all the new OIDs are correct? You'll need a postgres 13 server running locally.

File: test/sqllogictest/chbench.slt (hunk 1 above)
Comment (sploiselle) [reply test/sqllogictest/chbench.slt:315]: ty for pointing this out. it's a known issue (#4171), so I simply changed the table's column type to express the type that it used to resolve to (`integer`).

**Comment by sploiselle** [comment]
@mjibson ty for the catalog check command. It caught an issue with the `oid`s so was very germane. I hadn't realized there were both typed (`int2abs`) and untyped (`abs`) function names in PG, and had the wrong OID for the name function name we intended to use.

**Review by mjibson — APPROVED** [review]
Glad it caught something!

**Review by benesch — APPROVED** [review]
I hope you weren't waiting for my approval but if you were this looks awesome.

**Pull request merged**